core/src/main/java/com/spotify/metrics/tags/EnvironmentTagExtractor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.metrics.tags;
+
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Extract tags from the system environment variables.
+ *
+ */
+public class EnvironmentTagExtractor implements TagExtractor {
+
+    /**
+     * Prefix of environment variable that adds additional tags.
+     */
+    public static final String FFWD_TAG_PREFIX = "FFWD_TAG_";
+
+    public static Map<String, String> environmentTags;
+
+    public EnvironmentTagExtractor() {
+        this(Suppliers.ofInstance(System.getenv()));
+    }
+
+    public EnvironmentTagExtractor(Supplier<Map<String, String>> enviromentSupplier) {
+        this.environmentTags = filterEnvironmentTags(enviromentSupplier.get());
+    }
+
+    @Override
+    public Map<String, String> addTags(Map<String, String> tags) {
+        final Map<String, String> extractedTags = new HashMap<>(tags);
+        extractedTags.putAll(environmentTags);
```

Comment:
Does this means environment tags are overriding explicitly given tags?

Reply:
Yes. I will update the java doc to reflect this.